App/src/features/ranking/modal/charts/Spider.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Created by jules on 20/05/2017.
+ */
+
+import React, { Component, PropTypes } from 'react';
+import {
+    Dimensions,
+    View
+} from 'react-native';
+import LoadingIndicator from 'react-native-spinkit';
+import { Radar } from 'react-native-pathjs-charts';
+import _ from "lodash";
+import * as Intra from "../../../../api/intra";
+
+export class Spider extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.isMount = false;
+        this.state = {
+            modules: {},
+        };
+    }
+
+    async componentWillMount() {
+        this.isMount = true;
+        const { modules } = await Intra.fetchMarks(this.props.student.login);
+
+        this.isMount && this.setState({ modules: modules })
+    }
+
+    render() {
+
+        if (!this.state.modules.length) {
+            return (
+                <View style={{ alignItems: 'center', marginTop: 30 }}>
+                    <LoadingIndicator
+                        isVisible={!this.state.modules.length}
+                        color="#FFFFFF"
+                        type="Bounce"
+                        size={100}
+                    />
+                </View>
+            )
+        }
+
+        const rawGrades = _.map(this.state.modules, (module) => module.grade);
```

Comment:
Replace lodash `_.map` and `_.filter` for native ones ? You could also indeed chains `map` and `filter` wherever it's possible.